src/Orleans/Async/TaskExtensions.cs

```diff
@@ -285,39 +285,36 @@ internal static Task<T> ConvertTaskViaTcs<T>(Task<T> task)
 
             if (task.Status == TaskStatus.RanToCompletion)
```

Comment:
`RanToCompletion` is only true when the task *successfully* completed. Hence this replacement isn't equivalent to the original code.